packages/map-template/src/components/OpeningHours/OpeningHours.jsx

```diff
@@ -0,0 +1,141 @@
+import { memo, useCallback, useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import './OpeningHours.scss';
+import { ReactComponent as ClockIcon } from '../../assets/clock-light.svg';
+import { ReactComponent as ChevronDownIcon } from '../../assets/chevron-down.svg';
+import { ReactComponent as ChevronUpIcon } from '../../assets/chevron-up.svg';
+
+OpeningHours.propTypes = {
+    openingHours: PropTypes.object,
+    isAmFormat: PropTypes.bool,
+    isMondayFirstDayOfTheWeek: PropTypes.bool
+};
+
+/**
+ * 
+ * @param {object} props
+ * @param {object} props.openingHours // Opening hours data
+ * @param {boolean} [props.isAmFormat] // Whether to display time in AM/PM format
+ * @param {boolean} [props.isMondayFirstDayOfTheWeek] // Whether Monday is the first day of the week
+ * @returns 
+ */
+function OpeningHours({ openingHours, isAmFormat = false, isMondayFirstDayOfTheWeek = true }) {
+    const { t } = useTranslation();
+    const [isExpanded, setIsExpanded] = useState(false);
+    // Get real current day (0-6, Sunday is 0)
+    const currentDay = new Date().getDay();
+    // Create weekdays array with display order using useMemo
+    const weekdays = useMemo(() => {
+        const days = [
```

Comment:
We may be able eliminate the need for the translation for weekdays in all supported language by using toLocaleString:

![image](https://github.com/user-attachments/assets/2e3498c3-30f9-467e-9729-2f03240bfc6f)

And yes, it is Friday! 🎈